tests/pos-macros/quote-sym-newboundedtype/Macro_1.scala

```diff
@@ -0,0 +1,22 @@
+//> using options -experimental -Yno-experimental
+import scala.quoted.*
+
+inline def testMacro = ${ testImpl }
+
+transparent inline def transparentTestMacro = ${ testImpl }
+
+def testImpl(using Quotes): Expr[Object] = {
+    import quotes.reflect.*
+
+    def makeType(owner: Symbol): Symbol =
+      Symbol.newBoundedType(owner, "mytype", Flags.EmptyFlags, TypeBounds.lower(TypeRepr.of[String]), Symbol.noSymbol)
```

Comment:
Would be good to have some tests with type lambdas as bounds, that is the equivalent of:
```scala
type Foo = [X] =>> Int
type Bla >: Nothing <: [X] =>> Int
type Bar >: [X] =>> Int <: [X] =>> Int
```
It's also important that error-checking runs the same bound-checks than regular code:
```scala
type Foo1 >: Int <: String // error
type Foo2 >: [X] =>> Int <: Any // error: kind mismatch
```